examples/tests/client-tests.js

```diff
@@ -0,0 +1,173 @@
+    let paragraphs = section.querySelectorAll('p');
+    let as = section.querySelectorAll('a');
+    let images = as[0].querySelectorAll('img');
+    let deletes = section.querySelectorAll('div');
+
+    assert.ok(
+      section.classList.contains('picture'),
+      'A `section` must have the class `picture`.',
+    );
+
+    assert.strictEqual(
+      paragraphs.length,
+      1,
+      'There must be one `<p>` element in a `section`.',
+    );
+    assert.strictEqual(
+      paragraphs[0].className,
+      'title',
+      'The first paragraph must have the class `title`',
+    );
+
+    assert.strictEqual(
+      as.length,
+      1,
+      'There must be one `<a>` element in a `section`.',
+    );
+    assert.ok(
+      as[0].classList.contains('img'),
+      'The `<a>` element must have the class `img`.',
+    );
+    assert.strictEqual(
+      as[0].href,
+      data[0].file,
+      'The `<a>` element must have its `href` attribute set to the good value.',
+    );
+
+    assert.strictEqual(
+      images.length,
+      1,
+      'In a `<a>` element there must be only one image.',
+    );
+    assert.strictEqual(
+      images[0].src,
+      data[0].file,
+      'The `<img>` element must have its `src` attribute set to the link of the image.',
+    );
+    assert.strictEqual(
+      images[0].alt,
+      data[0].title,
+      'The `<img>` element must have its `alt` attribute set to the title of the image.',
+    );
+
+    assert.strictEqual(
+      deletes.length,
+      1,
+      'There must be only one `<div>` that displays an X in the `section`.',
+    );
+    assert.ok(
+      deletes[0].classList.contains('delete'),
+      'The `div` must have the class `delete`.',
+    );
+    assert.strictEqual(
+      deletes[0].textContent,
+      'X',
+      'The content of the `div` is `X`.',
+    );
+    assert.strictEqual(
+      deletes[0].dataset.id,
+      String(data[0].id),
+      'The `dataset.id` attribute of the `div` must be set to the `id` of the image.',
+    );
+    assert.strictEqual(
+      deletes[0].onclick,
+      window.forTesting.requestDelete,
+      'When clicking the `X`, the `requestDelete` function must run.',
+    );
+
+    // with an array of two pictures
+    data = [{
+      id: 1,
+      file: 'http://placebear.com/400/200',
+      title: 'A bear',
+    },
+    {
+      id: 2,
+      file: 'https://placebear.com/200/200',
+      title: 'Another bear',
+    }];
+    window.forTesting.putPicturesInPage(data);
+
+    assert.strictEqual(elM.childElementCount, (2+baseMainChildrenNumber), 'After passing two image entity to the function, there must be two sections.');
+
+    listOfSections = elM.querySelectorAll('section.picture');
+    section = listOfSections[1];
+    paragraphs = section.querySelectorAll('p');
+    as = section.querySelectorAll('a');
+    images = as[0].querySelectorAll('img');
+    deletes = section.querySelectorAll('div');
+
+    assert.strictEqual(listOfSections.length, 2);
```

Comment:
it's better to actually repeat the messages from above because if any of these assertions fail, the developer won't be told why, what's going on